### PR TITLE
Update bufbuild plugin references

### PR DESCRIPTION
**Pull request by mfridman**
Hey there, I noticed this project is referencing older Remote Plugins. There are upcoming [Remote Plugin changes](https://buf.build/blog/remote-packages-remote-plugins-approaching-v1) and this PR fixes the references to avoid any issues.

Older plugins on [buf.build](https://buf.build) will be removed in favor of Buf-managed plugins, which are managed by @bufbuild to ensure they work[^1] for the entire community.

In summary, the changes in this PR are as follows:

- Update older plugins to reference the new plugins in `connector/connect/common/src/main/buf.gen.yaml`
- Run `./dev/connect-gen-protos.sh` to generate the files

[^1]: Buf has an open-source [bufbuild/plugins](https://github.com/bufbuild/plugins) repository that builds, tests, patches and automates publishing popular Protobuf plugins. All plugins in that repository are kept up-to-date and published to [buf.build/plugins](https://buf.build/plugins)

### CODE REVIEW

File: connector/connect/common/src/main/buf.gen.yaml [hunk 1]
```diff
@@ -16,21 +16,21 @@
 #
 version: v1
 plugins:
-  - remote: buf.build/protocolbuffers/plugins/cpp:v3.20.0-1
```
Comment (mfridman) [inline connector/connect/common/src/main/buf.gen.yaml:19]: Newer versions of the `buf` CLI use the `plugin` key for both local and remote plugin references.

Note, I've also bumped the versions to the **latest available**, but that may not be what's desired.

File: connector/connect/common/src/main/buf.gen.yaml [hunk 2]
```diff
@@ -16,21 +16,21 @@
...
+  - plugin: buf.build/protocolbuffers/cpp:v22.3
```
Comment (mfridman) [inline connector/connect/common/src/main/buf.gen.yaml:19]: Newer versions of the `buf` CLI use the `plugin` key for both local and remote plugin references.

Note, I've also bumped the versions to the **latest available**, but that may not be what's desired.

File: connector/connect/common/src/main/buf.gen.yaml [hunk 3]
```diff
@@ -16,21 +16,21 @@
...
     out: gen/proto/ruby
    # Building the Python build and building the mypy interfaces.
-  - remote: buf.build/protocolbuffers/plugins/python:v3.19.3-1
+  - plugin: buf.build/protocolbuffers/python:v22.3
     out: gen/proto/python
-  - remote: buf.build/grpc/plugins/python:v1.47.0-1
+  - plugin: buf.build/grpc/python:v1.54.1
     out: gen/proto/python
-  - name: mypy
+  - plugin: buf.build/community/nipunn1313-mypy:v3.4.0
```
Comment (mfridman) [inline connector/connect/common/src/main/buf.gen.yaml:34]: I presume the mypy plugin is in referenced to the one maintained here: 

https://github.com/nipunn1313/mypy-protobuf

If so, this is now also a Remote Plugin available at:

https://buf.build/community/nipunn1313-mypy

**Comment by srowen** [comment]
Looks good, if you can resolve the conflicts and tests pass. File a little JIRA and update the title if you please too https://spark.apache.org/contributing.html

**Pull request closed without merging**